tests/python/pants_test/util/test_contextutil.py

```diff
@@ -29,6 +29,17 @@
 
 class ContextutilTest(unittest.TestCase):
 
+  @contextmanager
+  def setup_non_hermetic_environment(self):
```

Comment:
This naming seems mildly confusing to me. It seems like this is more of a "fake user environment" or something. The environment we're simulating isn't necessarily a non-hermetic one, just one that happens to have a user, I think?

Reply:
Agreed - good point. Renamed to `ensure_user_defined_in_environment`.